### ISSUE TITLE
Modify curlify through the Plugin system?

### ISSUE THREAD

**Issue by lcofre** [swagger-api/swagger-ui#4143]
First of all I would like to particularly thank you for the well thought design, the Plugin System in particular is really simple and a clever way to provide customization.

Explaining my issue, for a problematic endpoint on my definition I have to send multi-line text/plain on the  body. In the code, curlify.js sanitizes the body removing newlines:

```javascript
if ( request.get("body") ){
    if(type === "multipart/form-data" && request.get("method") === "POST") {
      // ...
    } else {
      curlified.push( "-d" )
      curlified.push( JSON.stringify( request.get("body") ).replace(/\\n/g, "") )
    }
}
```
But I would like to send the body as is and add "$" for curl to process it properly:
```javascript
if ( request.get("body") ){
    if(type === "multipart/form-data" && request.get("method") === "POST") {
      // ...
    } else {
      curlified.push( "-d" )
      if(type === "text/plain") {
        curlified.push( "$" )
        curlified.push( request.get("body") )
      } else
        curlified.push( JSON.stringify( request.get("body") ).replace(/\\n/g, "") )
    }
}
```

@shockey 
Can I achieve this through the plugin system? The change qualifies maybe as a pull-request?

Thanks in advance for your care and patience!

**Comment by shockey** [comment]
Hi @lcofre!

> First of all I would like to particularly thank you for the well thought design. In particular the Plugin System is really simple and a clever way to provide customization.

Thanks for the kind words :smile: we're always excited to see folks leveraging the plugin system to achieve more with Swagger.

> Can I achieve this through the plugin system? Or maybe the change qualifies as a pull-request?

This would be possible, but difficult, through the plugin system as it stands... for two reasons:

1. the Curl component is [tightly coupled to `curlify.js`](https://github.com/swagger-api/swagger-ui/blob/master/src/core/components/curl.jsx#L3), so you can't just override the curlify function
2. curlify's logic is contained in one function, so you can't just tweak one part of its logic, you'd need to provide a full custom curlify function.

Basically, if you had to do this today, you would use `wrapComponent` on Curl and manually replace what goes into the `textarea`. Pretty rough!

We can do better. Here's how I'd do it:

1. Expose the `curlify` function through the system `fn` interface, as part of a new `curl` core plugin
2. Add a `wrapFn` system interface that will allow you to modify curlify's behavior
3. Break `curlify`'s interior into smaller subfunctions, and add a provider interface that allows you to override _small parts_ of curlify's logic without having to reimplement the whole thing.

If you feel comfortable making the changes, that would be awesome, and I'd help you in the process any way I can... but if it sounds complex, I can do it, it will just take longer :smile:
